extension/httpforwarder/factory_test.go

```diff
@@ -32,4 +39,66 @@ func TestFactory(t *testing.T) {
 	require.Equal(t, configmodels.Type(expectType), cfg.Type())
 	require.Equal(t, ":6060", cfg.Ingress.Endpoint)
 	require.Equal(t, 10*time.Second, cfg.Egress.Timeout)
+
+	tests := []struct {
+		name           string
+		config         *Config
+		wantErr        bool
+		wantErrMessage string
+	}{
+		{
+			name:           "Default config",
+			config:         cfg,
+			wantErr:        true,
+			wantErrMessage: "'egress.endpoint' config option cannot be empty",
+		},
+		{
+			name:    "Invalid config",
+			config:  &Config{Egress: confighttp.HTTPClientSettings{Endpoint: "123.456.7.89:9090"}},
+			wantErr: true,
```

Comment:
nit, error message to explain why this is invalid?

Reply:
Added.